Assets/Scripts/Model/Graph.cs

```diff
@@ -21,5 +21,35 @@ internal void OnDeserializedMethod(StreamingContext context)
       if (edges == null)
         edges = new List<Edge>();
     }
+
+    public IList<Edge> FindEdgesByNodes(Node node1, Node node2)
+    {
+      var argSet = new HashSet<Node> {node1, node2};
+      return (
+        from edge in edges
+        where argSet.Contains(edge.nodeFrom) && argSet.Contains(edge.nodeTo)
+        select edge
+      ).ToList();
+    }
+
+    public IList<Edge> FindNodeEdges(Node node) =>
+      edges.Where(e => e.from == node.id || e.to == node.id).ToList();
+
+    public IDictionary<HashSet<Node>, List<Edge>> GetEdgesGroupedByNodes()
```

Comment:
It's getting to the point where we should be considering custom collection classes

Reply:
Yup